lib/dash/content_protection.js

```diff
@@ -252,16 +252,165 @@ shaka.dash.ContentProtection.parseFromRepresentation = function(
+* @property {number} length
+*   Size of the record in bytes.
+* @property {string} value
+*   Record content.
+*/
+shaka.dash.ContentProtection.PlayReadyRecord;
+
+/**
+ * Enum for PlayReady record types.
+ * @enum {number}
+ */
+shaka.dash.ContentProtection.PLAYREADY_RECORD_TYPES = {
+  RIGHTS_MANAGEMENT: 0x001,
+  RESERVED: 0x002,
+  EMBEDDED_LICENSE: 0x003,
+};
+
+
+/**
+* @param {Uint16Array} recordData
+* @param {number} recordCount
+* @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+* @private
+*/
+shaka.dash.ContentProtection.parseMsProRecords_ = function(
+  recordData, recordCount) {
+  let head = 0;
+  let records = [];
+  for (let i = 0; i < recordCount; i++) {
+    const recordRaw = recordData.subarray(head);
+    if (recordRaw.length > 1) {
+      const type = recordRaw[0];
+      const length = recordRaw[1];
+      const offset = 2;
+      const charCount = length / 2;
+      const end = charCount + offset;
+      // subarray end is exclusive
+      const rawValue = recordRaw.subarray(offset, end);
+      const value = shaka.util.StringUtils.fromUTF8(rawValue);
+      records.push({
+        type: type,
+        length: length,
+        value: value,
+      });
+      head = end;
+    }
+  }
+  return records;
+};
+
+
+/**
+* Based off getLicenseServerURLFromInitData from dash.js
+* https://github.com/Dash-Industry-Forum/dash.js
+*
+* @param {Uint16Array} bytes
+* @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+* @private
+*/
+shaka.dash.ContentProtection.parseMsPro_ = function(bytes) {
+  if (bytes.length < 2) {
+    return [];
+  }
+  const recordCount = bytes[2];
+  const recordData = bytes.subarray(3);
+  const ContentProtection = shaka.dash.ContentProtection;
+  return ContentProtection.parseMsProRecords_(recordData, recordCount);
+};
+
+
+/**
+* @param {!Element} xml
+* @return {string}
+* @private
+*/
+shaka.dash.ContentProtection.getLaurl_ = function(xml) {
+  const laurlNodes = xml.getElementsByTagName('LA_URL');
+  if (laurlNodes && laurlNodes.length > 0) {
+    const laurlNode = laurlNodes[0];
+    if (laurlNode.hasChildNodes()) {
+      const laurl = laurlNode.childNodes[0].nodeValue;
+      if (laurl) {
+        return laurl;
+      }
+    }
+  }
+  return '';
+};
+
+
+/**
+* Gets a PlayReady license URL from a content protection element
+* containing a PlayReady Header Object
+*
+* @param {shaka.dash.ContentProtection.Element} element
+* @return {string}
+*/
+shaka.dash.ContentProtection.getPlayReadyLicenseURL = function(element) {
+  const proNode = shaka.util.XmlUtils.findChild(element.node, 'mspr:pro');
+  if (!proNode) {
+    return '';
+  }
+  const ContentProtection = shaka.dash.ContentProtection;
+  const PLAYREADY_RECORD_TYPES = ContentProtection.PLAYREADY_RECORD_TYPES;
+  const bytes = shaka.util.Uint8ArrayUtils.fromBase64(proNode.textContent);
+  const records = ContentProtection.parseMsPro_(new Uint16Array(bytes.buffer));
+  const record = records.filter(function(record) {
```

Comment:
We are moving away from using `function`. Please use an arrow-function instead.

Reply:
Will fix.